MonstieWash/Assets/Scripts/Items/ItemLinker.cs

```diff
@@ -78,20 +77,26 @@ private void OnApplicationQuit()
 /// </summary>
     private void SaveItem()
     {
-        linkObject.Reset();
-        if (link == LinkMethod.Group)
+		linkObject.Reset();
+		if (link == LinkMethod.Group)
         {
             foreach (Transform child in transform)
             {
                 linkObject.positions.Add(FlipItemPos(child));
                 linkObject.rotations.Add(FlipItemRot(child));
+                linkObject.velocity.Add(FlipItemVel(child.GetComponent<Rigidbody2D>().velocity));
+                linkObject.angularVelocity.Add(FlipItemAng(child.GetComponent<Rigidbody2D>().angularVelocity));
+
                 linkObject.itemData.Add(new List<object>(child.GetComponent<ITaskScript>().SaveData()));
             }
         }
         else
         {
             linkObject.positions.Add(FlipItemPos(transform));
             linkObject.rotations.Add(FlipItemRot(transform));
+            linkObject.velocity.Add(FlipItemVel(GetComponent<Rigidbody2D>().velocity));
+            linkObject.angularVelocity.Add(GetComponent<Rigidbody2D>().angularVelocity);
+
             linkObject.itemData.Add(new List<object>(m_taskScript[0].SaveData()));
         }
         linkObject.loadOnEnable = true;
```

Comment:
The main bug I could find is that the following code that gets run when a bone is unstuck is not being completely applied to the linked bone. (Code from StuckItem.cs)

![image](https://github.com/user-attachments/assets/f6b47c59-1f94-4756-ab20-a628588e0537)

This means that if you pull out a bone and then throw it and switch scenes before it gets deactivated, it will fly off into narnia and never get deactivated. This is just because things like the rigidbody bodytype and OOBCheckRoutine aren't being run for the linked bone. So in short if you could edit this section of code to include all this stuff if possible that would fix things.

Reply:
Fiiiiiiiiiixed